lib/smart_answer/calculators/next_steps_for_your_business_calculator.rb

```diff
@@ -14,7 +14,16 @@ class NextStepsForYourBusinessCalculator
                   :business_premises
 
     def grouped_results
-      filtered_results.group_by { |result| result["topic"] }
+      results = {}
+      groups.each do |group|
+        results[group] = filtered_results.select { |result| result["group"] == group }.group_by { |result| result["topic"] }
+      end
+
+      results
+    end
+
+    def groups
+      filtered_results.collect { |result| result["group"] }.uniq!
```

Comment:
Nit: You can simplify this to:

```ruby
def grouped_results
  grouped_results = filtered_results.group_by { |result| result["group"] }
  
  grouped_results.transform_values do |results|
    results.group_by { |result| result["topic"] }
  end
end
```

Reply:
Thank you, @theseanything! `transform_values` was exactly what I was looking for as a Ruby rookie – have updated now.